syntax = "proto3";

message Stock {
  string exchange = 1;
  string code = 2;
  string full_name = 3;
  string industry = 4;
  string area = 5;
  int64 list_date = 6;
}

message StocksResponse {
  repeated Stock stocks = 1;
}

message StockDailySummary{
  int64 date = 1;
  double open = 2;
  double close = 3;
  double high = 4;
  double low = 5;
  double last_day_close = 6;
  double volume = 7;
  double turnover = 8;
}

message StockHistDailySummaries {
  Stock stock = 1;
  repeated StockDailySummary daily_summary = 2;
}

message StockPoint {
  int32 stock_index = 1;
  double stock_price = 2;
}

message StockPoints {
  repeated StockPoint stock_points = 1;
}
